cmdstanpy/model.py

```diff
@@ -297,6 +297,67 @@ def src_info(self) -> Dict[str, Any]:
             get_logger().debug(e)
             return result
 
+    def format_model(
+        self,
+        save: bool = False,
+        canonicalize: Union[bool, str, List[str]] = False,
+        *,
+        unsafe: bool = False,
+    ) -> None:
+        """
+        Run stanc's auto-formatter on the model code. Either saves directly
+        back to the file or prints for inspection
+
+
+        :param save: If True, save the updated code to disk, rather than
+            printing it. By default False
+        :param canonicalize: Whether or not the compiler should 'canonicalize'
+            the Stan model, removing things like deprecated syntax. Default is
+            False. If True, all canonicalizations are run. If it is a list of
+            strings, those options are passed to stanc (new in Stan 2.29)
+        :param unsafe: If True, do not create stanfile.bak backups before
+            writing to the file. Only do this if you're sure you have other
+            copies of the file or are using a version control system like Git.
+        """
+        if self.stan_file is None or not os.path.isfile(self.stan_file):
+            raise ValueError("No Stan file found for this module")
+        try:
+            cmd = (
+                [os.path.join(cmdstan_path(), 'bin', 'stanc' + EXTENSION)]
+                # handle include-paths, allow-undefined etc
+                + self._compiler_options.compose_stanc()
+                + [self.stan_file]
+            )
+
+            if canonicalize:
+                if isinstance(canonicalize, list):
```

Comment:
Should we also check for tuple or collections.abc.Sequence?

Reply:
I think we can actually expand it all the way to `collections.abc.Iterable`

Reply:
I think Sequence is more suitable?

Or what where the "rules"?

Either is fine by me